sw/device/silicon_creator/lib/sigverify/sphincsplus/test/wots_test.c

```diff
@@ -0,0 +1,86 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/wots.h"
+
+#include <stdint.h>
+
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/hash.h"
+#include "sw/device/silicon_creator/lib/sigverify/sphincsplus/params.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kSpxWotsMsgBytes = ((kSpxWotsLen1 * kSpxWotsLogW + 7) / 8),
+};
+
+static void pk_from_sig_test(const uint8_t *sig, const uint8_t *msg,
+                             const spx_ctx_t *ctx, spx_addr_t *addr,
+                             uint32_t *expected_pk) {
+  uint32_t actual_pk[kSpxWotsPkWords];
+  CHECK(wots_pk_from_sig(sig, msg, ctx, addr, actual_pk) == kErrorOk);
+  if (expected_pk != NULL) {
+    CHECK_ARRAYS_EQ(actual_pk, expected_pk, kSpxWotsPkWords);
+  }
+}
+
+bool test_main() {
+  LOG_INFO("Starting WOTS test...");
+
+  spx_ctx_t ctx = {
+      .pub_seed = {0xefbeadde},
+  };
+  CHECK(spx_hash_initialize(&ctx) == kErrorOk);
+  LOG_INFO("Setup complete.");
+
+  // Simple test:
+  //   sig = {0, 1, 2, 3, ... }
+  //   msg = { ..., 3, 2, 1, 0}
+  //   addr = {0xdeadbeef, 0xdeadbeef,...}
+  uint8_t sig[kSpxWotsBytes];
+  for (size_t i = 0; i < kSpxWotsBytes; i++) {
+    sig[i] = i & 255;
+  }
+  uint8_t msg[kSpxWotsMsgBytes];
+  for (size_t i = 0; i < kSpxWotsMsgBytes; i++) {
+    msg[i] = (kSpxWotsMsgBytes - i) & 255;
+  }
+  spx_addr_t addr = {.addr = {0xdeadbeef}};
+  // Note: this public key is based on the sphincs-shake-128s parameter set and
+  // will not work for other parameter sets.
```

Comment:
nit: Is it possible to add a reference to where we got this from or instructions on how to calculate this value?